main/resource-access.bs

```diff
@@ -5,28 +5,65 @@ Issue: Write introduction to the Authenticated Resource Access section.
 
 ## Hypertext Transfer Protocol ## {#http}
 
-Issue: Write Hypertext Transfer Protocol section.
-
-Draft:
-A Solid data pod MUST implement HTTP/1.1 Message Syntax and Routing [[!RFC7230]]
-
-Draft:
-A Solid data pod MUST implement HTTP/1.1 Semantics and Content [[!RFC7231]].
-
-Draft:
-A Solid data pod MUST implement HTTP/1.1 Authentication [[!RFC7235]].
-
-Draft:
-A Solid data pod SHOULD implement HTTP/1.1 Conditional Requests [[!RFC7232]].
-
-Draft:
-A Solid data pod SHOULD implement HTTP/1.1 Range Requests [[!RFC7233]].
-
-Draft:
-A Solid data pod SHOULD implement HTTP/1.1 Caching [[!RFC7234]].
-
-Draft:
-A Solid data pod MAY implement HTTP/2 [[!RFC7540]].
+### Background and Need ### {#http-need}
+<em>This section is non-normative.</em>
+
+Solid clients and servers need to exchange data securely over the Internet,
+and they do so using the Web protocol HTTP.
+This section describes in detail
+which parts of HTTP must be implemented by clients and servers.
+
+### Required server-side implementation ### {#http-server}
+
+A [=data pod=] MUST be an HTTP/1.1 server [[!RFC7230]][[!RFC7231]].
+It SHOULD additionally be an HTTP/2 server [[!RFC7540]]
+to improve performance,
+especially in cases where individual clients
+are expected to send high numbers of successive requests.
+
+A data pod SHOULD use the `https` URI scheme
+in order to secure the connection between clients and servers.
+When both `http` and `https` are supported,
+all `http` URIs MUST redirect to their `https` counterparts
+using a response with a `301` status code and a `Location` header.
```

Comment:
Have we agreed on that as an upgrade mechanism? Not e.g. [RFC2817](https://tools.ietf.org/html/rfc2817)

Reply:
I think so, yes. The 301 is an important semantic signal too. Also there's HTTP/2. Do you see any reason for using RFC2817 instead?

Reply:
No, I'm actually not very knowledgeable about it, I only know that RFC2817 was made for exactly this case and IIRC @timbl has been talking about it.

Also, it doesn't seem very settled what the semantics of `301` in relation to RDF, even though it has been [discussed for a long time](https://blog.ldodds.com/2007/03/07/the-semantics-of-301-moved-permanently/). It was [raised in the TAG](https://www.w3.org/2007/awwsw/tracker/issues/1) but doesn't seem to have a conclusion.

I suppose *we* could say somewhere that `301` means `owl:sameAs` :-)

Reply:
OK, we can reconsider in the future if needed.

Reply:
>When both `http` and `https` are supported

If only `http` is supported, the rest doesn't apply.

Hence, at least other upgrade mechanisms like RFC2817 can still be used.